staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -624,7 +666,9 @@ func (r *Reflector) list(stopCh <-chan struct{}) error {
 // After receiving a "Bookmark" event the reflector is considered to be synchronized.
 // It replaces its internal store with the collected items and
 // reuses the current watch requests for getting further events.
-func (r *Reflector) watchList(stopCh <-chan struct{}) (watch.Interface, error) {
+func (r *Reflector) watchList(ctx context.Context) (watch.Interface, error) {
```

Comment:
The elephant in the room which does not get addressed by this PR is the [ListWatcher API](https://github.com/kubernetes/kubernetes/blob/a499facee693a1a83daadb82d88f7b51d324ffc5/staging/src/k8s.io/client-go/tools/cache/listwatch.go#L46-L50). The implementations all use `context.TODO`. They get called here:
https://github.com/kubernetes/kubernetes/blob/a499facee693a1a83daadb82d88f7b51d324ffc5/staging/src/k8s.io/client-go/tools/cache/reflector.go#L677

In this PR, that code has a proper context, but cannot pass it on. Let's address that in a follow-up PR, similar to https://github.com/kubernetes/kubernetes/pull/129109?

Reply:
cc @p0lyn0mial

Reply:
I've addressed this in https://github.com/kubernetes/kubernetes/pull/129125. But that's a massive PR that I need to split up properly. The main benefit of doing the full exercise is that there will be no more surprises later on. Decisions like "this code does not need a context" could turn out to be incorrect when later it is found that the code directly or indirectly calls code which does need a context after all.